CONFIGURATION.md

```diff
@@ -192,10 +192,131 @@ c.IPKernelApp.extensions = ['rucio_jupyterlab.kernels.ipython']
+          def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+              self.enable_auth_state = True
+
+          def exchange_token(self, token):
+              params = {
+                  'client_id': self.client_id,
+                  'client_secret': self.client_secret,
+                  'grant_type': 'urn:ietf:params:oauth:grant-type:token-exchange',
+                  'subject_token': token,
+                  'scope': 'openid profile',
+                  'audience': 'rucio'
+              }
+              response = requests.post(self.token_url, data=params)
+              rucio_token = response.json()['access_token']
+              return rucio_token
+        
+          async def pre_spawn_start(self, user, spawner):
+              auth_state = await user.get_auth_state()
+              pprint.pprint(auth_state)
+              if not auth_state:
+                  # user has no auth state
+                  return
+            
+              # define token environment variable from auth_state
+              spawner.environment['RUCIO_ACCESS_TOKEN'] = self.exchange_token(auth_state['access_token'])
+              spawner.environment['EOS_ACCESS_TOKEN'] = auth_state['access_token']
+
+      # set the above authenticator as the default
+      c.JupyterHub.authenticator_class = RucioAuthenticator
+
+      # enable authentication state
+      c.GenericOAuthenticator.enable_auth_state = True
+
+      if 'JUPYTERHUB_CRYPT_KEY' not in os.environ:
+          warnings.warn(
+              "Need JUPYTERHUB_CRYPT_KEY env for persistent auth_state.\n"
+              "    export JUPYTERHUB_CRYPT_KEY=$(openssl rand -hex 32)"
+          )
+          c.CryptKeeper.keys = [os.urandom(32)]
+```
+
+3. Add the configuration parameters for the custom authenticator to the `hub.config`:
+
+```yaml
+hub:
+  config:
+    RucioAuthenticator:
+      client_id: <your-client-id>
+      client_secret: <your-client-secret>
+      authorize_url: <your-auth-url>
+      token_url: <your-token-url>
+      userdata_url: <your-userinfo-url>
+      username_key: preferred_username
+      scope:
+        - openid
+        - profile
+        - email
+```
+
+4. Add the required extension parameters to the `singleuser.extraEnv`:
+
+```yaml
+singleuser:
+  extraEnv:
+    RUCIO_MODE: "replica"
+    RUCIO_WILDCARD_ENABLED: "1"
+    RUCIO_BASE_URL: "<your-rucio-url>"
+    RUCIO_AUTH_URL: "<your-rucio-auth-url>"
+    RUCIO_WEBUI_URL: "<your-rucio-ui-url>"
+    RUCIO_DISPLAY_NAME: "<your-rucio-instance-display-name>"
+    RUCIO_NAME: "<your-rucio-instance-name>"
+    RUCIO_SITE_NAME: "<your-rucio-instance-site-name>"
+    RUCIO_OIDC_AUTH: "env"
+    RUCIO_OIDC_ENV_NAME: "RUCIO_ACCESS_TOKEN"
+    RUCIO_DEFAULT_AUTH_TYPE: "oidc"
+    RUCIO_OAUTH_ID: "<your-rucio-oauth-id>" # audience
+    RUCIO_DEFAULT_INSTANCE: "<your-rucio-instance-name>""
+    RUCIO_DESTINATION_RSE: "EOS RSE"
+    RUCIO_RSE_MOUNT_PATH: "/eos/eos-rse"
+    RUCIO_PATH_BEGINS_AT: "4"
```

Comment:
You could specify that this number makes reference to the amount of child paths in the  `extraVolumen.hostPath.path` argument, that makes reference to the FUSE mounted volumen.

Reply:
Good point, will add it! Thanks!